### ISSUE TITLE
how to get a youtube download url bypass IP access restrictions

### ISSUE THREAD

**Issue by kaiweizhang** [gantt/downloadyoutube#69]
when i use the YouTubeExtractor of frostwire to get the youtube download url,but i can't be used by other ip(youtube return 403).I have found some other website can get the url by web server,also the url includes ip param,but their can be use to download everywhere.I wonder how they do it.can you give me some solution,thanks!

**Comment by gantt** [comment]
What's that "other website"? I don't think you can get YouTube URLs without
limitations (IP, time).

On Thursday, January 14, 2016, kaiweizhang <notifications@github.com> wrote:

> when i use the YouTubeExtractor of frostwire to get the youtube download
> url,but i can't be used by other ip(youtube return 403)I have found some
> other website can get the url by web server,also the url includes ip
> param,but their can be use to download everywhereI wonder how they do itcan
> you give me some solution,thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gantt/downloadyoutube/issues/69>.
>


-- 
Ionut Alex Chitu
http://googlesystem.blogspot.com

**Comment by kaiweizhang** [comment]
I found Youtube URLs has limitations(IP),some song video has limit,like this video
https://www.youtube.com/watch?v=bqP-CVhtQfc
I get the download url but can't use in other ip(return 403)
some other website can get the right url like "http://keepvid.com/"

**Comment by gantt** [comment]
You're right, but I don't know how they obtain those links. I'll keep investigating

**Comment by kaiweizhang** [comment]
Ok,If you have any idea,please share with me,thank you very much!

**Comment by raszpl** [comment]
got a quick look at the links they generate
they seem to push ipv6, for example ip=2a03:8180:1001:16a::8ee1
maybe ipv6 confuses google and lets you download from different ipv4
